.travis.yml

```diff
@@ -32,11 +35,18 @@ script:
         --env-file <(env | grep -iE 'DEBUG|NODE_|ELECTRON_|YARN_|NPM_|CI|CIRCLE|TRAVIS|APPVEYOR_|CSC_|_TOKEN|_KEY|AWS_|STRIP|BUILD_') \
         --env ELECTRON_CACHE="/root/.cache/electron" \
         --env ELECTRON_BUILDER_CACHE="/root/.cache/electron-builder" \
+        --env X_RESET_UID=$UID \
         -v ${PWD}:/project \
         -v ~/.cache/electron:/root/.cache/electron \
         -v ~/.cache/electron-builder:/root/.cache/electron-builder \
+        -v $HOME/.npm:/root/.npm \
+        -v $HOME/.yarn-cache:/root/.yarn-cache \
         electronuserland/builder:wine \
-        /bin/bash -c "yarn --link-duplicates --pure-lockfile && yarn release --linux --win"
+        /bin/bash -c "\
+            yarn --link-duplicates --pure-lockfile &&\
+            yarn release --linux --win &&\
+            chown -R \$X_RESET_UID . /root \
```

Comment:
I suggest to change UID only for problematic files, not for all. If we have issues with electron zip files in the cache, better to change permissions only for these files.

Reply:
Any file created inside docker will be created as root:root, and is likely to cause issues if it needs to be changed in some way outside docker. We found no way to identify which ones needed to be changed (it might be something inside node_modules due to a new package being installed since last build).
This was the single thing we changed to get rid of the permissions issue when trying to do a simple "ls -la dist/win-unpacked" ("locales" and "resources" were not accessible).

Reply:
In our case we could not know in advance where the problems would surface. It happened to be in "dist/win-unpacked", on both "locales" and "resources" folder.
Basically, any file/folder created by docker sitting in a shared volume is going to appear as having been created by root once docker ends.

Reply:
Travis should cache only ZIP files, `locales` or `resources` should not cause issues. Need to investigate why issue occurred.